pkg/util/validation/limits_map.go

```diff
@@ -9,15 +9,19 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-// LimitsMap is a generic map that can hold either float64 or int as values.
-type LimitsMap[T float64 | int] struct {
+// LimitsMap is a flag.Value implementation that looks like a generic map, holding float64s, ints, or strings as values.
+type LimitsMap[T float64 | int | string] struct {
```

Comment:
So far, the only place where we really use map flags is in Limits, that's why this was called LimitsMap. But, nothing in this file is really specific to limits.

This new use case isn't really a limit, but I felt it gives a better configuration experience to be consistent with existing flags here, rather than invent a second map format that behaves differently.

I've kept the name `LimitsMap` and package as i feel it's _close enough_ to being descriptive and would muddy the diff to include renames/moves. I also think there's a valid case for moving this to `dskit`'s `flagext` package in the future anyway, so it didn't make sense to do a big rename yet.

I'm happy to change the name, or explore moving this to dskit, if reviewers would prefer. Or, this can be left to a subsequent PR.